src/gc/impl/conservative/gc.d

```diff
@@ -2078,6 +2078,135 @@ struct Gcx
     ToScanStack!(ScanRange) toscanPrecise;
 
     /**
+     * Finds the offset between an index and the next set bit, up to our max.
+     *
+     * Returns: The offset between index and the next bit,
+     *          or -1 if no bits are set between index and max
+     */
+    int bitScanForwardOffset(size_t* bitArray, size_t index, size_t max) nothrow
```

Comment:
> I'm not sure I understand. bitIndex is calculated with just a bit shift and testBit is a bitwise comparison and a bit shift. Aren't those fast operations?

They are recalculated for each call of bitScanForwardOffset, even though they are likely to be known in the previous call.

> I'm having trouble finding it right now, but I read that the log2 function was a constant time function

Could be constant, but still expensive. Aren't you just calculating `position = index & 63`?